src/components/FavoriteIcon.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { TouchableOpacity } from 'react-native'
+import Favorite from '@api/mwro/favorite'
+import { Heart } from 'lucide-react-native'
+import colors from '@ui/config/colors'
+
+type Props = {
+  isFavorite?: boolean
+  id: string
+  handleRefresh: () => void
+}
+
+export const handleToggleFavorite = async ({
+  isFavorite,
+  id,
+  handleRefresh
+}: Props) => {
+  isFavorite ? await Favorite.unfavorite(id) : await Favorite.favorite(id)
+  handleRefresh()
+}
+
+export default function FavoriteIcon({ isFavorite, id, handleRefresh }: Props) {
```

Comment:
Sempre que for passar o atributo de um objeto como prop, melhor passar o objeto inteiro. Nesse caso é loja, então ao invés de `{ isFavorite, id, handleRefresh }`, passar `{ store, handleRefresh }`.

Isso é importante por alguns motivos:

- Se precisarmos usar mais alguma propriedade de loja no futuro, fica fácil de fazer essa alteração (e essas coisas acontecem com bastante frequência)
- reduz a quantidade de props
- evita criar mais coisas em memória (especialmente nos casos que as propriedades sao objetos)
- deixa explicito de onde essas propriedades vem (id de quem, isFavorite de quem? etc)
- Não é o nosso caso, mas muitas empresas trabalham com ferramentas de tracking/analíticas no front, ter todas as propriedades da entitade em mãos facilita nessas horas também